### ISSUE TITLE
Large displacement by decelerate plugin on performance slowdowns

### ISSUE THREAD

**Issue by SukantPal** [davidfig/pixi-viewport#271]
## Brief 🎢 

The `DeceleratePlugin` relies on the elapsed time delta being a small value, so it can be treated as a _dt_. This assumption fails to realize when the browser cannot keep up at 60 FPS, causing the decelerate plugin to stretch itself (in time) and cause large displacements.

What do I mean by "stretching"? The velocity curve for deceleration at 60 FPS (purple) and 30 FPS (black) is shown below. Notice how the 30 FPS curve takes 2x longer to decay at to the same level? That means the `DecelaratePlugin` is not FPS-independent, and that's a problem.

<img width="2035" alt="Screen Shot 2020-11-25 at 9 09 47 AM" src="https://user-images.githubusercontent.com/22450567/100238262-fd86b700-2efd-11eb-86ed-3cdf9e648a22.png">

Another example is where the velocity is 82 and the delta is 30ms. The displacement is the area under the curve. The actual displacement should be the area under the purple curve. But since we approximate the velocity as constant per frame, the displacement is the area under the red curve. The difference - is the area between the red and purple curves. This error remains small then the time deltas are small. To fix it for larger deltas, we need a formula to calculate the area under the purple curve.

<img width="479" alt="Screen Shot 2020-11-25 at 9 16 04 AM" src="https://user-images.githubusercontent.com/22450567/100239000-dc729600-2efe-11eb-9d8d-acea03c8797f.png">

### Calculating the actual displacement needed in a frame

<img width="657" alt="Screen Shot 2020-11-25 at 9 28 27 AM" src="https://user-images.githubusercontent.com/22450567/100240459-97e7fa00-2f00-11eb-99da-723499aefad9.png">

### Updating velocity

<img width="279" alt="Screen Shot 2020-11-25 at 9 38 27 AM" src="https://user-images.githubusercontent.com/22450567/100241810-1db87500-2f02-11eb-9f22-baa46d205204.png">

### User adjustment

(you can skip this section as it just discusses how to tweak the friction for the same effect as the current approximation)

<img width="788" alt="Screen Shot 2020-11-25 at 10 25 58 AM" src="https://user-images.githubusercontent.com/22450567/100247712-a0dcc980-2f08-11eb-89b3-78de4b768bdf.png">

## TLDR;

If you want to use the "fluctuation immune" formula, I can make a PR. Let me know what you think.

**Comment by davidfig** [comment]
Yes, please do so. That might explain some jerking bugs we saw in Phase as well.
